src/predict.rs

```diff
@@ -113,137 +113,32 @@ pub trait Dim {
   const H: usize;
 }
 
-pub struct Block4x4;
-
-impl Dim for Block4x4 {
-  const W: usize = 4;
-  const H: usize = 4;
-}
-
-pub struct Block8x8;
-
-impl Dim for Block8x8 {
-  const W: usize = 8;
-  const H: usize = 8;
-}
-
-pub struct Block16x16;
-
-impl Dim for Block16x16 {
-  const W: usize = 16;
-  const H: usize = 16;
-}
-
-pub struct Block32x32;
-
-impl Dim for Block32x32 {
-  const W: usize = 32;
-  const H: usize = 32;
-}
-
-pub struct Block64x64;
-
-impl Dim for Block64x64 {
-  const W: usize = 64;
-  const H: usize = 64;
-}
-
-pub struct Block4x8;
-
-impl Dim for Block4x8 {
-  const W: usize = 4;
-  const H: usize = 8;
-}
-
-pub struct Block8x4;
-
-impl Dim for Block8x4 {
-  const W: usize = 8;
-  const H: usize = 4;
-}
-
-pub struct Block8x16;
-
-impl Dim for Block8x16 {
-  const W: usize = 8;
-  const H: usize = 16;
-}
-
-pub struct Block16x8;
-
-impl Dim for Block16x8 {
-  const W: usize = 16;
-  const H: usize = 8;
-}
-
-pub struct Block16x32;
-
-impl Dim for Block16x32 {
-  const W: usize = 16;
-  const H: usize = 32;
-}
-
-pub struct Block32x16;
-
-impl Dim for Block32x16 {
-  const W: usize = 32;
-  const H: usize = 16;
-}
-
-pub struct Block32x64;
-
-impl Dim for Block32x64 {
-  const W: usize = 32;
-  const H: usize = 64;
-}
-pub struct Block64x32;
-
-impl Dim for Block64x32 {
-  const W: usize = 64;
-  const H: usize = 32;
-}
-
-pub struct Block4x16;
-
-impl Dim for Block4x16 {
-  const W: usize = 4;
-  const H: usize = 16;
-}
-
-pub struct Block16x4;
-
-impl Dim for Block16x4 {
-  const W: usize = 16;
-  const H: usize = 4;
-}
-
-pub struct Block8x32;
-
-impl Dim for Block8x32 {
-  const W: usize = 8;
-  const H: usize = 32;
-}
-
-pub struct Block32x8;
-
-impl Dim for Block32x8 {
-  const W: usize = 32;
-  const H: usize = 8;
+macro_rules! block_dimension {
+  ($W:expr, $H:expr) => {
+    paste::item! {
```

Comment:
Can you explain what would be "item" here? Is it everything inside { ... } next to "item!"?

Reply:
I see, so in our case we use 'paste' with 'item' for copying declarations, not an expression.
I plan to use 'paste' to add rectangular tx functions, at least for declaration part, with 'item' then.
Do you think we can try even more tricks to shorten pub fn predict_intra<'a>() as well?
If possible, same idea can be applied to rect tx functions dispatcher, i.e. forward_transform() and inverse_transform_add().